2023-01-09 07:09:32 info: ----------------------------------------------
2023-01-09 07:09:32 info: >>> Running Unit Test `ex1-giveCompliment` <<<
2023-01-09 07:09:32 info: ----------------------------------------------
2023-01-09 07:09:37 error: *** Unit Test Error Report ***

✕ giveCompliment should have all TODO comments removed
✕ giveCompliment should give a random compliment: You are `compliment`, `name`!
  Expected: You are awesome, HackYourFuture!
  Received: You are awesome, HackYourFuture


Command failed: npx jest ex1-giveCompliment.test.js --colors --reporters="C:\Users\aoali\desktop\HYF_PROJECT\week6\Homework-class42\test-runner\CustomReporter.js"
2023-01-09 07:31:08 info: ----------------------------------------------
2023-01-09 07:31:08 info: >>> Running Unit Test `ex1-giveCompliment` <<<
2023-01-09 07:31:08 info: ----------------------------------------------
2023-01-09 07:31:13 error: *** Unit Test Error Report ***

✕ giveCompliment should have all TODO comments removed


Command failed: npx jest ex1-giveCompliment.test.js --colors --reporters="C:\Users\aoali\desktop\HYF_PROJECT\week6\Homework-class42\test-runner\CustomReporter.js"
2023-01-09 07:32:28 info: ----------------------------------------------
2023-01-09 07:32:28 info: >>> Running Unit Test `ex1-giveCompliment` <<<
2023-01-09 07:32:28 info: ----------------------------------------------
2023-01-09 07:32:33 error: *** Unit Test Error Report ***

✕ giveCompliment should have all TODO comments removed


Command failed: npx jest ex1-giveCompliment.test.js --colors --reporters="C:\Users\aoali\desktop\HYF_PROJECT\week6\Homework-class42\test-runner\CustomReporter.js"
2023-01-09 07:38:32 info: ----------------------------------------------
2023-01-09 07:38:32 info: >>> Running Unit Test `ex1-giveCompliment` <<<
2023-01-09 07:38:32 info: ----------------------------------------------
2023-01-09 07:38:36 info: All unit tests passed.
2023-01-09 07:38:42 info: All steps were completed successfully
2023-01-09 07:59:00 info: ----------------------------------------
2023-01-09 07:59:00 info: >>> Running Unit Test `ex2-dogYears` <<<
2023-01-09 07:59:00 info: ----------------------------------------
2023-01-09 07:59:04 info: All unit tests passed.
2023-01-09 07:59:11 info: All steps were completed successfully
2023-01-09 08:11:04 info: --------------------------------------------
2023-01-09 08:11:04 info: >>> Running Unit Test `ex4-shoppingCart` <<<
2023-01-09 08:11:04 info: --------------------------------------------
2023-01-09 08:11:04 warn: A unit test file was not provided.
2023-01-09 08:11:12 info: All steps were completed successfully
2023-01-09 08:12:50 info: --------------------------------------------
2023-01-09 08:12:50 info: >>> Running Unit Test `ex4-shoppingCart` <<<
2023-01-09 08:12:50 info: --------------------------------------------
2023-01-09 08:12:50 warn: A unit test file was not provided.
2023-01-09 08:12:57 info: All steps were completed successfully
2023-01-09 08:18:37 info: -------------------------------------------
2023-01-09 08:18:37 info: >>> Running Unit Test `ex7-mindPrivacy` <<<
2023-01-09 08:18:37 info: -------------------------------------------
2023-01-09 08:18:37 warn: A unit test file was not provided.
2023-01-09 08:18:44 info: All steps were completed successfully
2023-01-09 08:24:10 info: ------------------------------------------------
2023-01-09 08:24:10 info: >>> Running Unit Test `ex5-shoppingCartPure` <<<
2023-01-09 08:24:10 info: ------------------------------------------------
2023-01-09 08:24:10 warn: A unit test file was not provided.
2023-01-09 08:24:17 info: All steps were completed successfully
2023-01-10 05:21:35 info: -----------------------------------------
2023-01-10 05:21:36 info: >>> Running Unit Test `ex6-totalCost` <<<
2023-01-10 05:21:36 info: -----------------------------------------
2023-01-10 05:21:36 warn: A unit test file was not provided.
2023-01-10 05:21:43 info: All steps were completed successfully
2023-01-10 06:54:45 info: -------------------------------------------
2023-01-10 06:54:45 info: >>> Running Unit Test `ex3-tellFortune` <<<
2023-01-10 06:54:45 info: -------------------------------------------
2023-01-10 06:54:48 error: *** Unit Test Error Report ***

✕ tellFortune should tell the fortune by randomly selecting array values
  Expected: You will be a Developer in Istanbul, married to Jan with 1 kids.
  Received: You will be a Developer in Istanbul, married to Jan with 1 kid.


Command failed: npx jest ex3-tellFortune.test.js --colors --reporters="C:\Users\aoali\desktop\HYF_PROJECT\week6\Homework-class42\test-runner\CustomReporter.js"
2023-01-10 06:56:34 info: -------------------------------------------
2023-01-10 06:56:34 info: >>> Running Unit Test `ex3-tellFortune` <<<
2023-01-10 06:56:34 info: -------------------------------------------
2023-01-10 06:56:37 info: All unit tests passed.
2023-01-10 06:56:41 info: All steps were completed successfully
2023-02-09 12:22:33 info: ---------------------------------------
2023-02-09 12:22:33 info: >>> Running Unit Test `ex1-johnWho` <<<
2023-02-09 12:22:33 info: ---------------------------------------
2023-02-09 12:22:39 error: *** Unit Test Error Report ***

Command failed: npx jest ex1-johnWho.test.js --colors
 FAIL  3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js
  getAnonName
    √ should exist and be executable (1 ms)
    × should have all TODO comments removed (2 ms)
    √ `getAnonName` should not contain unneeded console.log calls (1 ms)
    √ should call `new Promise()`
    √ should take a single argument
    √ `resolve()` should be called with a one argument
    √ `reject()` should be called with a one argument (1 ms)
    √ should resolve when called with a string argument (1 ms)
    √ should reject with an Error object when called without an argument

  ● getAnonName › should have all TODO comments removed

    expect(received).toBeFalsy()

    Received: true

      103 | function testTodosRemoved(getSource) {
      104 |   test('should have all TODO comments removed', () => {
    > 105 |     expect(/\bTODO\b/.test(getSource())).toBeFalsy();
          |                                          ^
      106 |   });
      107 | }
      108 |

      at Object.toBeFalsy (test-runner/unit-test-helpers.js:105:42)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 8 passed, 9 total
Snapshots:   0 total
Time:        0.878 s
Ran all test suites matching /ex1-johnWho.test.js/i.
2023-02-09 12:23:13 info: ---------------------------------------
2023-02-09 12:23:13 info: >>> Running Unit Test `ex1-johnWho` <<<
2023-02-09 12:23:13 info: ---------------------------------------
2023-02-09 12:23:17 info: All unit tests passed.
2023-02-09 12:23:23 info: All steps were completed successfully
2023-02-09 12:32:33 info: -------------------------------------------------
2023-02-09 12:32:33 info: >>> Running Unit Test `ex2-checkDoubleDigits` <<<
2023-02-09 12:32:33 info: -------------------------------------------------
2023-02-09 12:32:38 error: *** Unit Test Error Report ***

Command failed: npx jest ex2-checkDoubleDigits.test.js --colors
 FAIL  3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js
  checkDoubleDigits
    √ should exist and be executable (2 ms)
    √ should have all TODO comments removed (1 ms)
    √ `checkDoubleDigits` should not contain unneeded console.log calls (1 ms)
    √ should call new Promise()
    √ `resolve()` should be called with a one argument (1 ms)
    √ `reject()` should be called with a one argument
    √ should be a function that takes a single argument
    √ (9) should return a rejected promise with an Error object (2 ms)
    × (10) should return a promise that resolves to "This is a double digit number!" (2 ms)
    × (99) should return a promise that resolves to "This is a double digit number!" (1 ms)
    √ (100) should return a rejected promise with an Error object

  ● checkDoubleDigits › (10) should return a promise that resolves to "This is a double digit number!"

    expect(received).resolves.toEqual()

    Received promise rejected instead of resolved
    Rejected to value: [Error: Expected a double digit number but got 10]

      73 |     const promise = checkDoubleDigits(10);
      74 |     expect(promise).toBeInstanceOf(Promise);
    > 75 |     return expect(promise).resolves.toEqual(
         |            ^
      76 |       expect.stringContaining('This is a double digit number!')
      77 |     );
      78 |   });

      at expect (node_modules/expect/build/index.js:128:15)
      at Object.expect (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:75:12)

  ● checkDoubleDigits › (10) should return a promise that resolves to "This is a double digit number!"

    expect.assertions(3)

    Expected three assertions to be called but received two assertion calls.

      69 |
      70 |   test('(10) should return a promise that resolves to "This is a double digit number!"', () => {
    > 71 |     expect.assertions(3);
         |            ^
      72 |     expect(exported).toBeDefined();
      73 |     const promise = checkDoubleDigits(10);
      74 |     expect(promise).toBeInstanceOf(Promise);

      at Object.assertions (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:71:12)

  ● checkDoubleDigits › (99) should return a promise that resolves to "This is a double digit number!"

    expect(received).resolves.toEqual()

    Received promise rejected instead of resolved
    Rejected to value: [Error: Expected a double digit number but got 99]

      82 |     const promise = checkDoubleDigits(99);
      83 |     expect(promise).toBeInstanceOf(Promise);
    > 84 |     return expect(promise).resolves.toEqual(
         |            ^
      85 |       expect.stringContaining('This is a double digit number!')
      86 |     );
      87 |   });

      at expect (node_modules/expect/build/index.js:128:15)
      at Object.expect (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:84:12)

  ● checkDoubleDigits › (99) should return a promise that resolves to "This is a double digit number!"

    expect.assertions(2)

    Expected two assertions to be called but received one assertion call.

      79 |
      80 |   test('(99) should return a promise that resolves to "This is a double digit number!"', () => {
    > 81 |     expect.assertions(2);
         |            ^
      82 |     const promise = checkDoubleDigits(99);
      83 |     expect(promise).toBeInstanceOf(Promise);
      84 |     return expect(promise).resolves.toEqual(

      at Object.assertions (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:81:12)

Test Suites: 1 failed, 1 total
Tests:       2 failed, 9 passed, 11 total
Snapshots:   0 total
Time:        1.084 s
Ran all test suites matching /ex2-checkDoubleDigits.test.js/i.
2023-02-09 12:33:39 info: -------------------------------------------------
2023-02-09 12:33:39 info: >>> Running Unit Test `ex2-checkDoubleDigits` <<<
2023-02-09 12:33:39 info: -------------------------------------------------
2023-02-09 12:33:44 info: All unit tests passed.
2023-02-09 12:33:50 info: All steps were completed successfully
